Library/Source/NativeEngine.cpp

```diff
@@ -1357,13 +1325,7 @@ namespace babylon
 
             callbackPtr->Call({});
 
-            // recycle viewIds used as viewports
-            for (auto id : m_viewportIds)
-            {
-                m_viewidSet.Recycle(id);
-            }
-            m_viewportIds.clear();
-            m_currentBackbufferViewId = 0;
+            m_viewidSet.Clear();
```

Comment:
It's still very strange to me that the `NativeEngine` owns this and the `FrameBufferManager` just has a reference to it. This also introduces some weirdness in cases where `NativeEngine` doesn't own the frame loop, such as in XR. If this is the only place where the `NativeEngine` interacts with the `RecycleSet`, would it be better to give ownership of the `RecycleSet` back to the `FrameBufferManager` and add a `FrameBufferManager::Reset()` method?